config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/BundleEntityBuilder.java

```diff
@@ -34,11 +45,93 @@
         List<Entity> entities = new ArrayList<>();
         entityBuilders.forEach(builder -> entities.addAll(builder.build(bundle, bundleType, document)));
 
-        return bundleDocumentBuilder.build(document, entities);
+        Map<String, Entity> annotatedEntities = new HashMap<>();
+        // Filter the bundle to export only annotated entities
+        // TODO : Enhance this logic to support services and policies
+        final Map<String, Encass> encassEntities = bundle.getEntities(Encass.class);
+        encassEntities.entrySet().parallelStream().forEach(encassEntry -> {
+            Encass encass = encassEntry.getValue();
+            if (encass.getAnnotations() != null) {
+                encass.getAnnotations().forEach(entityAnnotation -> {
+                    if (entityAnnotation.getType().equals(ANNOTATION_TYPE_BUNDLE)) {
+                        final String annotatedEntityName = entityAnnotation.getName();
+                        final String annotatedEntityDesc = entityAnnotation.getDescription();
+                        final Entity filteredEntity = entities.parallelStream().filter(entity -> encassEntry.getKey().equals(entity.getName())).findAny().get();
+                        annotatedEntities.put(annotatedEntityName != null ? annotatedEntityName + '-' + bundleVersion : encassEntry.getKey() + '-' + bundleVersion, filteredEntity);
+                    }
+                });
+            }
+        });
+
+        if (annotatedEntities.size() > 0) {
+            Map<String, Element> annotatedElements = new HashMap<>();
+            annotatedEntities.entrySet().parallelStream().forEach(annotatedEntity -> {
+                List<Entity> entityList = getEntityDependencies(annotatedEntity.getValue(), entities, bundle);
+                LOGGER.log(Level.WARNING, "Annotated entity dependencies" + entityList);
+                if (EntityBuilder.BundleType.DEPLOYMENT == bundleType) {
+                    entityList.add(annotatedEntity.getValue());
+                }
+                annotatedElements.put(annotatedEntity.getKey(), bundleDocumentBuilder.build(document, entityList));
+            });
+            return annotatedElements;
+        }
+
+        return new HashMap<String, Element>() {
+            {
+                put(bundleName + '-' + bundleVersion, bundleDocumentBuilder.build(document, entities));
+            }
+        };
+    }
+
+    private List<Entity> getEntityDependencies(Entity entity, List<Entity> entities, Bundle bundle) {
+        List<Entity> entityDependenciesList = new ArrayList<>();
+        Map<Dependency, List<Dependency>> dependencyListMap = bundle.getDependencyMap();
+        if (dependencyListMap != null) {
+            Set<Map.Entry<Dependency, List<Dependency>>> entrySet = dependencyListMap.entrySet();
+            for (Map.Entry<Dependency, List<Dependency>> entry : entrySet) {
+                Dependency annotatedEntity = entry.getKey();
```

Comment:
i think annotatedEntity name is wrong, it should be used for first method argument